Feature/create message script

andnotand opened the pull request:
new branch

andnotand committed:
カクテル選択画面の検索と選択をするscriptを作成

andnotand committed:
カクテル選択画面のhtmlにタグ付けしてスクリプトが動く様に変更

pre-commit-ci[bot] committed:
chore(fix): Auto fixes from pre-commit.com hooks

For more information, see https://pre-commit.ci.

Ryo-7 commented on public/script.js:
```diff
@@ -0,0 +1,123 @@
+}
+
+function updatecontainersize(){
+    var buttonstyle = window.getComputedStyle(document.getElementById('button'), '');
```
```suggestion
    const buttonstyle = window.getComputedStyle(document.getElementById('button'), '');
```

javascriptでvarは非推奨になってます。
変数はletを使って欲しいし、今回は宣言後に書き換えないと思うので constで良いかなと思いました。

Ryo-7 commented on public/script.js:
```diff
@@ -0,0 +1,123 @@
+function updatecontainersize(){
+    var buttonstyle = window.getComputedStyle(document.getElementById('button'), '');
+    const height = buttonstyle.getPropertyValue('height');
+    const width = buttonstyle.getPropertyValue('width');
```
エディタでみると、この変数は使ってなさそうです。
使ってない変数はバグの原因になるので、消しておくと良いと思います。

Ryo-7 commented on public/script.js:
```diff
@@ -0,0 +1,123 @@
+                throw new Error('Network response was not ok');
+            }
+            //return response.json(); // JSON形式のレスポンスをパース
+            console.log(response);
```
これはデバッグ用の `console.log`だと思うので、消しておきましょう

他にもコメントアウトしてる不要なコードは削除しておくと、全体の行数が減ってみやすくなります。

今は使わないけど前使ってて、いつか使うかもしれないというコードはgitのコミット上に載るので、そこから引っ張り出せるという点でも、基本的にコードはコメントアウトではなく削除でいいと思います

Ryo-7 commented on public/script.js:
```diff
@@ -0,0 +1,123 @@
+            text += node.textContent.trim();
+        }
+    });
+    const b_text = text;//クリックしたボタンの文字
```
```suggestion
    const selectedCocktailMessage = text;//クリックしたボタンの文字
```
基本的に、javascriptの変数はlowerCamelCaseで書きます。
また、b_textだと何の変数か、将来の自分を含め分からなくなると思うので上記のようなのはどうでしょうか？ 🤔

Ryo-7 commented on src/components/CreateMessageSelect.tsx:
```diff
@@ -126,60 +124,63 @@ export const CreateMessageSelect = () => {
+          {/*</button>*/}
           <div class={checkLinkStyle}>
-            <a href="./check">確認</a>
+            <a href="/create/check">確認</a>
```
ここは僕の考慮ミスなんですが、今回のPRに不必要な変更が入ってしまっているので修正して欲しいです。 🙇

Ryo-7 commented on src/components/CreateMessageSelect.tsx:
```diff
@@ -14,8 +14,8 @@ const selectedStyle = css`
     margin: 0.5rem 0;
-    border-radius: 2rem;
-    align-items: center;
+    border-radius: 0.5rem;
```
この辺も不要な変更が入ってしまった気がします 🙇

Ryo-7 commented on src/components/CreateMessageSelect.tsx:
```diff
@@ -96,13 +96,10 @@ const checkLinkStyle = css`
- */
-
-export const CreateMessageSelect = () => {
+export function CreateMessageSelect() {
```
ここも不要な気がします 🙇

Ryo-7 reviewed (changes requested):
スクリプト追加ありがとうございます。

ちゃんと動作していて一気にアプリっぽくなっていい感じです。 👍 

コードを書く上でちょっと気になるところがあるので修正お願いします 🙇 
コメントつけてなくても同じような修正箇所がいくつかあるので全体をざっと確認してみて欲しいです。

5ouma commented on public/script.js:
```diff
@@ -0,0 +1,123 @@
+//tmp_list = ['apple', 'orange', 'grape', 'strawberry', 'apples'];//テスト用
```
Prettier等でコードをフォーマットしておくといいかも
インデントの数や波括弧前後のスペース等

yui-szk replied:
このコメントも要らなそう

yui-szk committed:
Fix words covered button (#27)

#25 
メッセージ確認画面で選択した言葉がボタンに被る問題を修正しました

5ouma commented on public/script.js:
```diff
@@ -0,0 +1,123 @@
+
+    if(created_sentence.includes(b_text)){
+        //nothing
+    }else if(document.getElementById("c_button").childElementCount >= 4){//配列の要素で判定
```
4つまでは許容するはず？

```suggestion
    }else if(document.getElementById("c_button").childElementCount > 4){//配列の要素で判定
```

5ouma commented on public/script.js:
```diff
@@ -0,0 +1,123 @@
+        });
+        //console.log(text2);//for test
+
+        if (text2.includes(search_kekka)) {
+            li.style.display = ''; // 検索文字列を含む場合は表示
+        } else {
+            li.style.display = 'none'; // 含まない場合は非表示
+        }
```
三項演算子を使うと綺麗になるかも

```suggestion
        li.style.display = text2.includes(search_kekka) ? "" : "none";
```

5ouma commented on public/script.js:
```diff
@@ -0,0 +1,123 @@
+    //console.log(b_text);
+
+
+    if(created_sentence.includes(b_text)){
+        //nothing
+    }else if(document.getElementById("c_button").childElementCount >= 4){//配列の要素で判定
+        alert("これ以上言葉を追加できません");
+    }else{
```
`//nothing`があるのであれば、以下に変えるといいかも

```suggestion
    if (document.getElementById("c_button").childElementCount >= 4) {
        alert("これ以上言葉を追加できません");
    } else if (!created_sentence.includes(b_text)) {
```

yui-szk commented on public/script.js:
```diff
@@ -0,0 +1,123 @@
+
+const container = document.getElementById('container');
+
+window.addEventListener('load', (event) => {
```
引数の`event`は使用してなさそうなので、削除してもいいかもしれません

5ouma commented on public/script.js:
```diff
@@ -0,0 +1,123 @@
+
+const container = document.getElementById('container');
+
+window.addEventListener('load', (event) => {
+    // ウィンドウが読み込まれた時の処理
+    fetch('/api/cocktail/all')
+        .then(response => {
+            if (!response.ok) {
+                throw new Error('Network response was not ok');
+            }
+            //return response.json(); // JSON形式のレスポンスをパース
+            console.log(response);
+        })
+        .then(data => {
+            console.log('Received data:', data); // 取得したデータをコンソールに出力
+            // ここで、データを使った他の処理を実行することができます
+        })
+        .catch(error => {
+            console.error('Error fetching data:', error); // エラーハンドリング
+        });
+});
```
この`addEventListner`は何もしていない？

yui-szk replied:
私も何してるかわからなかった...

yui-szk commented on public/script.js:
```diff
@@ -0,0 +1,123 @@
+                }
+            }
+            created_sentence = tmp;
+            //console.log("created_sentence:" + created_sentence);成功
```
必要ないコメントは削除しましょう

yui-szk commented on public/script.js:
```diff
@@ -0,0 +1,123 @@
+                                     .filter(node => node.nodeType === Node.TEXT_NODE)
+                                     .map(node => node.textContent.trim())
+                                     .join(' ');
+            //console.log(textContent); // クリックされた <li> 要素のテキストを表示
```
必要ないコメントは削除しましょう

yui-szk commented on public/script.js:
```diff
@@ -0,0 +1,123 @@
+                text2 += node.textContent.trim();
+            }
+        });
+        //console.log(text2);//for test
```
必要ないコメントは削除しましょう

yui-szk commented on public/script.js:
```diff
@@ -0,0 +1,123 @@
+    }else{
+        created_sentence.push(b_text);
+
+        //console.log(created_sentence);//fortest
```
必要ないコメントは削除しましょう

yui-szk commented on public/script.js:
```diff
@@ -0,0 +1,123 @@
+        }
+    });
+    const b_text = text;//クリックしたボタンの文字
+    //console.log(b_text);
```
必要ないコメントは削除しましょう

yui-szk commented on public/script.js:
```diff
@@ -0,0 +1,123 @@
+};
+
+
+document.addEventListener('DOMContentLoaded', function() {
```
```suggestion
document.addEventListener('DOMContentLoaded', () => {
```
アロー関数に統一した方がいいかも

5ouma committed:
refactor(backend): Refactor backend API (#29)

It doesn't return the `success` field to use `Response.status` and validate too many or duplicated items.

andnotand committed:
Update script.js

andnotand committed:
Update public/script.js

Co-authored-by: Souma <101255979+5ouma@users.noreply.github.com>

yui-szk commented on public/share.js:
作業内容が異なる場合はブランチを変更して作業してください

yui-szk commented on src/components/CreateMessageCheck.tsx:
```diff
@@ -96,6 +96,7 @@ const sendButtonStyle = css`
 export const CreateMessageCheck = () => {
   return (
     <WithHTML>
+
```
不必要な差分は出ないようにしてほしい

Ryo-7 committed:
javascriptの修正とrefの遷移先のパスの修正

Ryo-7 committed:
不要なデバッグログの削除と関数をアローに変更

Ryo-7 committed:
chore(fix): Auto fixes from pre-commit.com hooks

For more information, see https://pre-commit.ci.

Ryo-7 committed:
コメントの整理

Ryo-7 committed:
変数名の修正とhmtlの不要なボタン要素の削除

Ryo-7 committed:
Update public/script.js

Co-authored-by: Souma <101255979+5ouma@users.noreply.github.com>

Ryo-7 committed:
不要なコメントの削除

Ryo-7 committed:
不要なコメントの削除

Ryo-7 committed:
Merge branch 'main' into feature/create_message_script

yui-szk commented on src/components/CreateMessageSelect.tsx:
```diff
@@ -126,60 +125,61 @@ export const CreateMessageSelect = () => {
                 <span class="material-symbols-outlined">close</span>
               </button>
             </li>
+            */}
```
不要なコメントは削除をー

yui-szk commented on src/components/CreateMessageSelect.tsx:
```diff
@@ -126,60 +125,61 @@ export const CreateMessageSelect = () => {
-              秘密<small>シンガポール・スリング</small>
-            </li>
-          </ul>
+          {/*<button id="kennsaku_result">*/}
```
コメント削除

yui-szk commented on src/components/CreateMessageSelect.tsx:
```diff
@@ -126,60 +125,61 @@ export const CreateMessageSelect = () => {
+                秘密<small>シンガポール・スリング</small>
+              </li>
+            </ul>
+          {/*</button>*/}
```
コメント削除

Ryo-7 commented:
ここだけ検索窓が切れてるのが気になりました。
~~サクッと直せそうなら修正コミットを積んでもらって、厳しそうなら別タスクを切りましょう~~
![スクリーンショット 2024-09-04 14 31 48](https://github.com/user-attachments/assets/a71eb5c9-1e91-47cb-8790-3c9f52aedc72)

修正されてました 👍

Ryo-7 reviewed (dismissed):
動作に問題なさそう

LGTM 👍 

---
コメントを残す癖はめっちゃ大事です。

なんでも残せばいいってものじゃないのが難しいところですが、初見だとこのコードわからんだろうなってところに書くようにすると良いです。

次のPRはちょっとだけ意識してもらえると、良いコードになって最高だと思います 👍

yui-szk commented on src/components/CreateMessageSelect.tsx:
```diff
@@ -126,60 +125,61 @@ export const CreateMessageSelect = () => {
-              <input type="text" />
-              <button type="submit">
+              <input type="text" value ="" id="kennsaku" placeholder="ここに検索したい語句を入力してください" oninput="kennsaku_show()"/>
                 <span class="material-symbols-outlined">search</span>
```
```suggestion
              <button type="submit">
                <span class="material-symbols-outlined">search</span>
              </button>
```
スタイルがずれているので修正お願いします

ボタンを押すと検索リセットされるので、後でアイコン修正するかもしれない

andnotand committed:
検索ボタン修正

Co-authored-by: yui-szk <148746551+yui-szk@users.noreply.github.com>

andnotand committed:
CreateMessageSelect.tsx: comment delete

yui-szk reviewed (dismissed):
LGTM！

5ouma commented on src/components/CreateMessageSelect.tsx:
```diff
@@ -96,13 +97,10 @@ const checkLinkStyle = css`
- */
-
-export const CreateMessageSelect = () => {
+export function CreateMessageSelect() {
```
元に戻した方がよさそう

5ouma commented on src/components/CreateMessageSelect.tsx:
```diff
@@ -112,74 +110,62 @@ export const CreateMessageSelect = () => {
-            </li>
-          </ul>
+            <ul class={wordListStyle} id="kennsaku_result">
+              <li onclick="buttonclick(this)">
+                あなたを守りたい<small>ブルドッグ</small>
+              </li>
+              <li onclick="buttonclick(this)">
+                いつも2人で<small>サイドカー</small>
+              </li>
+              <li onclick="buttonclick(this)">
+                正しき心<small>ジンバッグ</small>
+              </li>
+              <li onclick="buttonclick(this)">
+                素晴らしい<small>アプリコット・クーラー</small>
+              </li>
+              <li onclick="buttonclick(this)">
+                冒険<small>アラウンド・ザ・ワールド</small>
+              </li>
+              <li onclick="buttonclick(this)">
+                偽りなき心<small>アラスカ</small>
+              </li>
+              <li onclick="buttonclick(this)">
+                幸せいっぱい<small>ウエディング・ベル</small>
+              </li>
+              <li onclick="buttonclick(this)">
+                晴れやかな心で<small>エバー・グリーン</small>
+              </li>
+              <li onclick="buttonclick(this)">
+                あなたに会いたい<small>ケーブルグラム</small>
+              </li>
+              <li onclick="buttonclick(this)">
+                華麗<small>コスモポリタン</small>
+              </li>
+              <li onclick="buttonclick(this)">
+                あるがままに<small>ジン・フィズ</small>
+              </li>
+              <li onclick="buttonclick(this)">
+                秘密<small>シンガポール・スリング</small>
+              </li>
+            </ul>
```
頭に空白が入ってます

5ouma commented on public/script.js:
```diff
@@ -0,0 +1,109 @@
+let created_sentence = [];
+
+window.addEventListener('load', () => {
+    // ウィンドウが読み込まれた時の処理
+    fetch('/api/cocktail/all')
+        .then(response => {
+            if (!response.ok) {
+                throw new Error('Network response was not ok');
+            }
+            return response.json();
+        })
+        .then(data => {
+            console.log('Received data:', data); // 取得したデータをコンソールに出力
+            // ここからデータのリストを作成
+            const liElements = document.getElementById("kennsaku_result");
+            for(i = 0; i < data.length; i++){
+                const li = document.createElement('li');
+                li.setAttribute('onclick', 'buttonclick(this)');
+                li.textContent = data[i].word;
+
+                const small = document.createElement('small');
+                small.textContent = data[i].name;
+
+                li.appendChild(small);
+                liElements.appendChild(li);
+            }
+        })
+        .catch(error => {
+            console.error('Error fetching data:', error); // エラーハンドリング
+        });
+});
```
[CreateMessageSelect.tsx](https://github.com/jigintern/online-2024-e/blob/c7c44feb1051b7f589409a507945dc20159e9e60/src/components/CreateMessageSelect.tsx)を下に置き換えると、APIから取ってきたデータを表示できるのですが…

```tsx
import { hc } from "@hono/hono/client";
import { css } from "@hono/hono/css";

import { WithHTML } from "../layout/WithHTML.tsx";
import { API } from "../api/mod.ts";
import { Cocktail } from "../api/utils/types.ts";

const client = hc<API>("http://0.0.0.0:8000/api");
const getAllCocktails = async (): Promise<Cocktail[]> => {
  const res = await client.cocktail.all.$get();
  if (!res.ok) throw new Error(await res.text());
  return await res.json();
};

const selectedStyle = css`
  height: 24vh;
  list-style: none;
  margin-top: 1.5rem;

  li {
    display: grid;
    grid-template-columns: 3fr 1fr;
    background-color: var(--color-base);
    font-size: 1rem;
    color: var(--color-main);
    padding: 0.25rem 0.75rem;
    margin: 0.5rem 0;
    border-radius: 2rem;
    align-items: center;

    button {
      justify-self: end;
      border: none;
      background-color: var(--color-base);
    }
  }
`;

const wordListContainerStyle = css`
  background-color: var(--color-base);
  border-radius: 1.25rem;

  form {
    display: flex;
    padding: 1rem;

    input {
      background-color: var(--color-main);
      color: var(--color-white);
      font-size: 1rem;
      width: 100%;
      border-radius: 0.5rem 0 0 0.5rem;
      padding: 0.75rem;
      border: none;
      outline: none;
    }

    button {
      border: none;
      background-color: var(--color-main);
      color: var(--color-white);
      border-radius: 0 0.5rem 0.5rem 0;
      padding-right: 0.5rem;
    }
  }
`;

const wordListStyle = css`
  list-style: none;
  color: var(--color-main);
  height: calc(100vh - 24vh - 9.25rem);
  overflow-y: scroll;

  li {
    display: grid;
    grid-template-columns: 5fr 2fr;
    font-size: 1rem;
    align-items: center;
    padding: 1rem;
    padding-left: 1.5rem;
    border-top: solid 1px var(--color-stroke);

    small {
      color: #7c7c7c;
      font-size: 0.75rem;
    }
  }
`;

const checkLinkStyle = css`
  position: absolute;
  bottom: -1.25rem;
  left: 50%;
  transform: translateX(-50%);

  a {
    background-color: var(--color-accent);
    width: 5rem;
    height: 5rem;
    border-radius: 100%;
    color: var(--color-white);
    text-decoration: none;
    font-size: 1.25rem;
    display: flex;
    align-items: center;
    justify-content: center;
  }
`;

/**
 * メッセージを選択する画面を返す関数
 */

export const CreateMessageSelect = async () => {
  return (
    <WithHTML>
      <link
        rel="stylesheet"
        href="https://fonts.googleapis.com/css2?family=Material+Symbols+Outlined:opsz,wght,FILL,GRAD@24,400,0,0"
      />
      <link
        rel="stylesheet"
        href="https://fonts.googleapis.com/css2?family=Material+Symbols+Outlined:opsz,wght,FILL,GRAD@24,400,0,0"
      />
      <div>
        <div>
          <ul class={selectedStyle}>
            <li>
              あなたを守りたい
              <button>
                <span class="material-symbols-outlined">close</span>
              </button>
            </li>
            <li>
              いつも2人で
              <button>
                <span class="material-symbols-outlined">close</span>
              </button>
            </li>
          </ul>
        </div>
        <div class={wordListContainerStyle}>
          <div>
            <form action="">
              <input type="text" />
              <button type="submit">
                <span class="material-symbols-outlined">search</span>
              </button>
            </form>
          </div>
          <ul class={wordListStyle}>
            {(await getAllCocktails()).map((cocktail: Cocktail) => (
              <li>
                {cocktail.word}
                <small>{cocktail.name}</small>
              </li>
            ))}
          </ul>
          <div class={checkLinkStyle}>
            <a href="./check">確認</a>
          </div>
        </div>
      </div>
    </WithHTML>
  );
};
```

Ryo-7 replied:
`.tsx`で書いた方が良さそうなので、@5ouma にこの部分の変更をお願いします。
ただ、このPRを入れないと辛いと思うので、このPRをマージした後で別PRとして出して欲しいです  🙏

5ouma commented on public/script.js:
```diff
@@ -0,0 +1,109 @@
+let created_sentence = [];
```
Prettier等でコードをフォーマットしておくといいかも
インデントの数や波括弧前後のスペース等

andnotand committed:
soumaの指摘箇所修正

andnotand committed:
Lint Checkでエラーが起きていたので修正

andnotand committed:
修正2

andnotand committed:
修正その３

andnotand committed:
format error

andnotand committed:
format error

andnotand committed:
Merge branch 'main' into feature/create_message_script

andnotand committed:
動作しないエラーを修正

andnotand committed:
script.js

andnotand committed:
script.js

andnotand committed:
createmessageselect

Ryo-7 reviewed (dismissed):
動作するのとCIが通っているのでLGTM

細かい気になるところは、最悪後で僕が直します

5ouma commented on public/share.js:
空ファイル？

5ouma commented on src/components/CreateMessageSelect.tsx:
```diff
@@ -1,5 +1,6 @@
 import { css } from "@hono/hono/css";
 import { WithHTML } from "../layout/WithHTML.tsx";
+//import {buttonclick, kennsaku_show} from "../../public/script.js"
```
これは消しましょう

yui-szk replied:
使わないやつは消してほしいな...

5ouma commented on src/components/CreateMessageCheck.tsx:
```diff
@@ -130,7 +130,6 @@ export const CreateMessageCheck = () => {
             </p>
           </div>
         </div>
-
```
元に戻した方がよさそう？

yui-szk commented on src/components/CreateMessageCheck.tsx:
```diff
@@ -139,6 +138,13 @@ export const CreateMessageCheck = () => {
+              気持ちを共有する
+            </a>
+          </div>
+
```
他機能の実装なので、入れないでほしいな...

送る、のボタンは既にあるから、要らない気もする

andnotand committed:
余計なコメントアウトの削除etc

yui-szk commented on src/components/CreateMessageSelect.tsx:
```diff
@@ -97,7 +99,7 @@ const checkLinkStyle = css`
 
 /**
- * メッセージを選択する画面を返す関数
+ * 作成するメッセージを選択する画面を返す関数
```
変更されてるので直してほしいです

andnotand committed:
コメントetcの修正

andnotand committed:
space

pre-commit-ci[bot] committed:
chore(fix): Auto fixes from pre-commit.com hooks

For more information, see https://pre-commit.ci.

Ryo-7 reviewed (dismissed):
動作する、不要なファイルはないことを確認

LGTM 👍

5ouma commented on public/script.js:
```diff
@@ -0,0 +1,103 @@
+        .join(" ");
+      liElement.remove();
+
+      tmp = [];
```
変数宣言を明示した方がよさそう

5ouma commented on src/components/CreateMessageSelect.tsx:
```diff
@@ -112,66 +113,59 @@ export const CreateMessageSelect = () => {
-            </li>
-          </ul>
+        <div id="clicked_button">
+          <ul class={selectedStyle} id="c_button"></ul>
```
`c_button`だと何のボタンが分かりづらいね

5ouma commented on public/script.js:
```diff
@@ -0,0 +1,103 @@
+
+  const liElements = document.querySelectorAll("#kennsaku_result li");
+  liElements.forEach((li) => {
+    let text2 = "";
```
`text1`が無いので`text`で良いのでは

Ryo-7 committed:
tmp配列をconstで宣言するように変更

Ryo-7 committed:
選択済みのカクテルメッセージを表示する領域のid名を修正

Ryo-7 committed:
変数名を変更

5ouma committed:
Merge branch 'main' into feature/create_message_script
